docs/advanced_topics.rst

```diff
@@ -1,114 +1,252 @@
+batch of data, and, if necessary, to prefetch. If the execution mode is set to asynchronous,
+this call returns immediately, without waiting for the results. This way, another task can be
+simultaneously executed. The data batch can be requested from DALI by calling share_outputs,
+which returns the result buffer. If the data batch is not yet ready, DALI will wait for it.
+The data is ready as soon as the :meth:`nvidia.dali.pipeline.Pipeline.share_outputs()``
+is complete. When the DALI buffers are no longer needed, because data was copied or has
+already been consumed, call :meth:`nvidia.dali.pipeline.Pipeline.release_outputs()` to return
+the DALI buffers for reuse in subsequent iterations.
+
+Built-in iterators use the second API to provide convenient wrappers for immediate use in
+Deep Learning Frameworks. The data is returned in the framework’s native buffers. The iterator’s
+implementation copies the data internally from DALI buffers and recycles the data by calling
+:meth:`nvidia.dali.pipeline.Pipeline.release_outputs()`.
+
+We recommend that you do not mix the  APIs. The APIs follow a different logic for the output
+buffer lifetime management, and the details of the process are subject to change without notice.
+Mixing the APIs might result in undefined behavior, such as a deadlock or an attempt to access
+an invalid buffer.
 
-The first API - :meth:`nvidia.dali.pipeline.Pipeline.run` method launches the DALI pipeline, executing prefetch iterations if necessary, waits until the first batch is ready and returns the resulting buffers. Buffers are marked as in-use untill the next call to `nvidia.dali.pipeline.Pipeline.run`. In many cases, it is wasteful as data is usually copied out to the native framework tensors after which they can be returned to DALI to be reused
+Sharding
+--------
 
-The second API, consisting of :meth:`nvidia.dali.pipeline.Pipeline.schedule_run`, :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` and :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` allows the user to explicitly manage the lifetime of the output buffers. The :meth:`nvidia.dali.pipeline.Pipeline.schedule_run` method instructs DALI to prepare the next batch of data, prefetching if necessary. If the execution mode is set to asynchronous, this call returns immediately, without waiting for the results, so another task can be executed in parallel. The data batch can be requested from DALI by calling `share_outputs`, which returns the result buffer. If it is not ready yet, DALI will wait for it. The data is ready as soon as the :meth:`nvidia.dali.pipeline.Pipeline.share_outputs` method returns. When DALI buffers are no longer needed, because data was copied or already consumed, :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` should be called to return DALI buffers to be reused in subsequent iterations.
+Sharding allows DALI to partition the dataset into nonoverlapping pieces on which each DALI pipeline
```

Comment:
Typo in `nonoverlapping`

Reply:
I can't see it. `nonoverlapping` sounds good.

Reply:
Yup, my bad.

Reply:
disjoint?